sorting/dnf_sort.c

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file
+ * @brief Program to perform [dutch national
+ * flag](https://en.wikipedia.org/wiki/Dutch_national_flag_problem) of a target
+ * value in a given {0, 1, 2} containing array.
```

Comment:
Provide a detailed description of the program/algorithm by using `@details`.
You can take information from Wikipedia and meld it here. 🙂